### ISSUE TITLE
Implement `entityCount` for subgraphs so that it affects subscriptions

### ISSUE THREAD

**Issue by Jannis** [graphprotocol/graph-node#618]
**Do you want to request a *feature* or report a *bug*?**

Enhancement (somewhere in the middle)

**What is the current behavior?**

The `entityCount` field is implemented as a computed field. Removing entities or adding new entities however does not trigger subscription updates, which we'll want.

**What is the expected behavior?**

The expected behavior is that, as the number of entities change, subscriptions for the "subgraph of subgraphs" are updated.

A potential solution would be to re-count the entities of a subgraph when applying entity operations—which should be fairly cheap—and then write the new value to an `entityCount` field in the subgraph entity.

**Comment by leodasvacas** [comment]
Seems we opened the same issue twice, so I'll close the other one and move the report here:

In PR https://github.com/graphprotocol/graph-node/pull/602 we introduced the concept of computed fields, which are fields in a schema that are not stored, for the subgraph of subgraphs so we could implement `entityCount`. However because the field is not stored, it cannot be subscribed to.

Supporting subscriptions would require a special listener to the store. If this turns out to be difficult or not a priority, we could at least return an error if someone tries to subscribe to it.
